CMakeLists.txt

```diff
@@ -73,11 +73,17 @@ else (DOXYGEN_FOUND)
 endif (DOXYGEN_FOUND)
 
 enable_language(CUDA)
-find_package(GSL)
-find_package(LAPACK)
-find_package(Casacore REQUIRED)
+
+# Global Libraries
 find_package(Boost REQUIRED)
+find_package(Casacore REQUIRED)
+find_package(LAPACK)
+find_package(GSL)
+
+# Local Libraries
+set(EIGEN3_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/external/eigen-3.3.90)
 find_package(Eigen3 REQUIRED)
```

Comment:
These two lines can be turned into a single `add_subdirectory(external/eigen-3.3.90)` instead, which in turn means you don't need to add the `cmake/FindEigen3.cmake` file to the repo. In the previous version you actually had `include_directories(external/eigen-3.3.90)` (and removed linking to `Eigen3::Eigen`) which also worked fine - but linking against a target is better.

The same doesn't work with RapidJSON's repository though because it doesn't define a `RapidJSON::RapidJSON` target (https://github.com/Tencent/rapidjson/issues/1687).

Reply:
First time I tried this I had some strange build errors but working now. The only significant difference between the two is that add_subdirectory also builds the eigen3 tests which ctest picks in the test runner.

The tests take a few minutes and flood the console with warnings which I'm not to fond of doing regularly, I suspect there may be a flag to exclude tests somewhere but maybe this is something better for future if I wanted to change the eigen build flags.